EventFiltering/PWGHF/HFFilter.cxx

```diff
@@ -766,10 +791,60 @@ struct HfFilter { // Main struct for HF triggers
   using BigTracksWithProtonPID = soa::Filtered<soa::Join<aod::BigTracksExtended, aod::TrackSelection, aod::pidTPCFullPr, aod::pidTOFFullPr>>;
 
   void process(aod::Collision const& collision,
+               aod::BCsWithTimestamps const&,
                HfTrackIndexProng2withColl const& cand2Prongs,
                HfTrackIndexProng3withColl const& cand3Prongs,
                BigTracksWithProtonPID const& tracks)
   {
+    if (applyML && ccdbTimestamp == 0 && onnxFiles[kD0] == "") {
+      thresholdBDTScores = {
+        thresholdBDTScoreD0ToKPi,
+        thresholdBDFScoreDPlusToPiKPi,
+        thresholdBDFScoreDSToPiKK,
+        thresholdBDFScoreLcToPiKP,
+        thresholdBDFScoreXicToPiKP};
+
+      onnxFiles = {
+        onnxFileD0ToKPiConf,
+        onnxFileDPlusToPiKPiConf,
+        onnxFileDSToPiKKConf,
+        onnxFileLcToPiKPConf,
+        onnxFileXicToPiKPConf};
```

Comment:
> We can define these two arrays outside the `init` and `process` functions, to avoid code duplication

yes, but seems we only can keep it in the init but outside the 'applyML' function. it will always be Initialized.